indico/modules/categories/forms.py

```diff
@@ -106,70 +112,138 @@ class CategorySettingsForm(IndicoForm):
-                    _('The Issuer ID is invalid or not linked to your credentials.')
+        if config.ENABLE_GOOGLE_WALLET:
+            enabled = self.google_wallet_mode.data == InheritableConfigMode.enabled
+            credentials = self.google_wallet_credentials.data
+            issuer_id = self.google_wallet_issuer_id.data
+
+            if (
+                credentials and
+                enabled and (
+                    self.category.google_wallet_settings.get('google_wallet_credentials') != credentials or
+                    self.category.google_wallet_settings.get('google_wallet_issuer_id') != issuer_id
                 )
-                return False
+            ):
+                res = GoogleWalletManager.verify_credentials(credentials, issuer_id)
+                if res == GoogleCredentialValidationResult.invalid:
+                    self.google_wallet_credentials.errors.append(
+                        _('The credentials could not be loaded.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.refused:
+                    self.google_wallet_credentials.errors.append(
+                        _('The credentials are invalid or have no access to the Google Wallet API.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.failed:
+                    self.google_wallet_credentials.errors.append(
+                        _('There was an error validating your credentials. Contact your Indico administrator for '
+                          'details.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.bad_issuer:
+                    self.google_wallet_issuer_id.errors.append(
+                        _('The Issuer ID is invalid or not linked to your credentials.')
+                    )
+                    return False
 
         return form_valid
 
+    def validate_apple_pass_certificate(self, field):
+        try:
+            x509.load_pem_x509_certificate(field.data.encode())
+            return True
+        except ValueError:
+            raise ValidationError(_('The provided Certificate.pem is malformed.'))
+
+    def validate_apple_pass_key(self, field):
+        try:
+            hazmat.primitives.serialization.load_pem_private_key(field.data.encode(), password=None)
+        except ValueError:
+            raise ValidationError(_('The provided Private.key is malformed.'))
+        except TypeError:
+            # This is because I provided an empty pass, just to check the Private.key is not malformed.
+            # Is there a better way to check it?
+            return True
+        except UnsupportedAlgorithm:
+            logger.warning('The provided Google Wallet private key type for category %s is not supported.',
+                           self.category.id)
```

Comment:
What's the rationale behind logging this fact? The validation error ensures that there will be no side-effects in the system.

Reply:
Providing a wrong private key is "rare" and probably the person is uploading it in a wrong format.
Logging the error will help us to recognise the user (from the category) and eventually give support.
If you don't want it just tell, I'll remove it.

Reply:
While I'm not the biggest fan of it, I think there is some merit in keeping it because it's interesting what people pasted in that form when this happens...

That said, personally I'd simply have kept it as an unhandled exception. As you say, it's very rare, so failing noisily wouldn't be a big deal. Anyway, I simply made the log message a bit more concise so I'd say it's fine to leave it as it is now.